indico_toolkit/snapshots/snapshot.py

```diff
@@ -0,0 +1,210 @@
+from __future__ import (
+    annotations,
+)  # from 3.10, don't need for same class reference in class method
+from typing import List
+import pandas as pd
+import json
+from json import JSONDecodeError
+from indico_toolkit import ToolkitInstantiationError, ToolkitInputError
+
+# TODO: add functionality for classification snapshots
+
+
+class Snapshot:
+    def __init__(
+        self,
+        path_to_snapshot: str,
+        text_col: str = None,
+        label_col: str = None,
+        file_name_col: str = None,
```

Comment:
we should also include the "row index" column. It will allow you to use the snapshot to label a teach task.